core/observers/STFLogger.cpp

```diff
@@ -23,76 +36,90 @@ namespace pegasus
             stf_writer_.setTraceFeature(stf::TRACE_FEATURES::STF_CONTAIN_RV64);
         }
 
-        // TODO: add support for memory records
+        // TODO: Add support for PTE
+        // stf_writer_.setTraceFeature(stf::TRACE_FEATURES::STF_CONTAIN_PTE);
 
         stf_writer_.setISA(stf::ISA::RISCV);
         stf_writer_.setHeaderPC(inital_pc);
         stf_writer_.finalizeHeader();
         recordRegState_(state);
     }
 
-    // METHODS
     void STFLogger::postExecute_(PegasusState* state)
     {
-        if (fault_cause_.isValid() || interrupt_cause_.isValid())
+        if (state->getNextPc() != state->getPrevPc() + state->getCurrentInst()->getOpcodeSize())
         {
-            return;
+            stf_writer_ << stf::InstPCTargetRecord(state->getNextPc());
+        }
```

Comment:
You should only use this record type for taken branches. Right now, this will also insert an `InstPCTargetRecord` for faults, interrupts, syscalls, etc.

Reply:
However, until those other cases are wired up you can replace this with a `ForcePCRecord` for all non-branch cases.